### ISSUE TITLE
Debugging javascript with Oh My Posh produces a weird debugger state and no output

### ISSUE THREAD

**Issue by bnm12** [microsoft/vscode#175233]
Type: <b>Bug</b>

Using WSL 2 (Ubuntu 22.04.2)
When I have the Oh My Posh theme in my .bashrc file and lunch a debug task that starts a npm task in the javascript debugging terminal it seems like the task executes before Oh My Posh does and gets cut off. The debugger starts, but no output prints to the terminal.
Also no breakpoints get hit.
Re-running the command that prints to the terminal manually after it's initalized (same terminal window) works as expected

Initial state
![image](https://user-images.githubusercontent.com/2268946/220912536-d99d49e2-0f50-40c8-9fdb-18af0d572329.png)

Re-running command manually
![image](https://user-images.githubusercontent.com/2268946/220913050-33c053f1-37ba-465c-badc-e0518ede4c2e.png)

My launch configuration is: 
```
{
  "name": "Test: TSNode Current File",
  "type": "node-terminal",
  "request": "launch",
  "command": "npm run mocha:local:glob ${relativeFile}",
  "env": {
    "LOGLEVEL": "debug"
  }
}
```

VS Code version: Code 1.75.1 (441438abd1ac652551dbe4d408dfcec8a499b8bf, 2023-02-08T21:32:34.589Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes
Remote OS version: Linux x64 5.10.16.3-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i7-13700K (24 x 3418)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|63.75GB (34.93GB free)|
|Process Argv|--file-uri vscode-remote://wsl%2Bubuntu-22.04/home/bnm12/enode-git/api/api.code-workspace --crash-reporter-id ae095441-8f6f-4bd0-b81e-e8e6ca5105a5|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu-22.04|
|OS|Linux x64 5.10.16.3-microsoft-standard-WSL2|
|CPUs|13th Gen Intel(R) Core(TM) i7-13700K (24 x 3417)|
|Memory (System)|54.93GB (51.15GB free)|
|VM|0%|
</details><details><summary>Extensions (44)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
Bookmarks|ale|13.3.1
postcss|css|1.0.9
better-cpp-syntax|jef|1.17.2
jupyter-keymap|ms-|1.0.0
remote-containers|ms-|0.275.1
remote-ssh|ms-|0.96.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.75.3
vscode-remote-extensionpack|ms-|0.24.0
remote-explorer|ms-|0.2.0
remote-server|ms-|1.0.0
vscode-serial-monitor|ms-|0.8.0
aws-toolkit-vscode|ama|1.63.0
vscode-tailwindcss|bra|0.9.7
doxdocgen|csc|1.4.0
vscode-eslint|dba|2.4.0
gitlens|eam|13.2.0
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.10.4
vscode-pull-request-github|Git|0.58.2
vscode-mocha-test-adapter|hbe|2.14.1
vscode-test-explorer|hbe|2.21.1
cmake-language-support-vscode|jos|0.0.9
vscode-docker|ms-|1.23.3
vscode-dotnet-runtime|ms-|1.6.0
isort|ms-|2022.8.0
python|ms-|2023.2.0
vscode-pylance|ms-|2023.2.40
jupyter|ms-|2023.1.2010391206
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.14
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
cmake-tools|ms-|1.13.45
cpptools|ms-|1.14.3
cpptools-extension-pack|ms-|1.3.0
test-adapter-converter|ms-|0.1.6
vscode-serial-monitor|ms-|0.8.0
typescript-mono-repo-import-helper|q|0.0.6
command-variable|rio|1.49.1
code-spell-checker|str|2.17.1
linear|str|0.0.3
cmake|twx|0.0.17

(3 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30669237
azure-dev_surveyone:30548225
pyindex848:30662994
nodejswelcome1cf:30587006
282f8724:30602487
pyind779:30662992
89544117:30613380
pythonsymbol12cf:30657549
vsctsb:30662444
pythonb192cf:30669361

```

</details>

**Comment by roblourens** [comment]
Is there anything we can do here @Tyriar @meganrogge?

**Comment by meganrogge** [comment]
a lot going on here 

> launch a debug task that starts an npm task in the javascript debugging terminal 

I think @connor4312 might know more

**Comment by connor4312** [comment]
That's fine and valid. The issue here seems solely that the text is injected before the shell(?) is ready and the ignored. Not sure whether that's entirely on us or not.

Here's the code where the terminal is started and program is written https://github.com/microsoft/vscode-js-debug/blob/bebebd6638df3bc884a3c8f785b4ec3f87f7c832/src/targets/node/terminalNodeLauncher.ts#L132-L146

@meganrogge do you think we should await the `terminal.processId` to avoid this, or any other recommendation?

**Comment by bnm12** [comment]
Would this mean that any sufficiently large/slow `.bashrc` file would cause this behavior? If so that should be pretty easy to set up a repro for using a wait of some kind shouldn't it?

**Comment by meganrogge** [comment]
Yes @connor4312 I think that would fix it. We do that in debug before sending text
https://github.com/microsoft/vscode/blob/1759ad36d2cb9ba9bdcec8866b35ee94b1f642ce/src/vs/workbench/api/node/extHostDebugService.ts#L117-L135

**Comment by connor4312** [comment]
Ah, I see, thanks. It's quite unfortunate that we have that one second pause if we need to `giveShellTimeToInitialize`. Is there any stronger signal we can use to avoid that artificial delay?

**Comment by meganrogge** [comment]
Oh, I don't think you'll need to do that. I would think just the `await terminal.processId` would be sufficient
